python/src/equistore/operations/join.py

```diff
@@ -101,11 +101,19 @@ def join(tensor_maps: List[TensorMap], axis: str):
             )
 
             for parameter, gradient in block.gradients():
+                if len(gradient.gradients_list()) != 0:
+                    raise NotImplementedError(
+                        "gradients of gradients are not supported"
+                    )
+
                 new_block.add_gradient(
                     parameter=parameter,
-                    data=gradient.data,
-                    samples=gradient.samples,
-                    components=gradient.components,
+                    gradient=TensorBlock(
+                        values=gradient.values,
+                        samples=gradient.samples,
+                        components=gradient.components,
+                        properties=gradient.properties,
```

Comment:
I thought this is not possible with the current API?

Reply:
Actually could gradient=gradient work in the above?

Reply:
no, because `gradient` is already owned by another block. We could do `gradient=gradient.copy()`, but that's going to be slower. 

I'll add a comment on the code explaining that.

Reply:
Actually, this code was slightly wrong, it should be using the same properties as the block, which might be different from the ones in `gradient.properties`.